packages/assets-controllers/src/MultichainBalancesController/BalancesTracker.ts

```diff
@@ -0,0 +1,139 @@
+import { Poller } from './Poller';
+
+type BalanceInfo = {
+  lastUpdated: number;
+  blockTime: number;
+};
+
+const BALANCES_TRACKING_INTERVAL = 5000; // Every 5s in milliseconds.
+
+export class BalancesTracker {
+  #poller: Poller;
+
+  #updateBalance: (accountId: string) => Promise<void>;
+
+  #balances: Record<string, BalanceInfo> = {};
+
+  constructor(updateBalanceCallback: (accountId: string) => Promise<void>) {
+    this.#updateBalance = updateBalanceCallback;
+
+    this.#poller = new Poller(
+      () => this.updateBalances(),
+      BALANCES_TRACKING_INTERVAL,
+    );
+  }
+
+  /**
+   * Starts the tracking process.
+   */
+  start(): void {
+    this.#poller.start();
+  }
+
+  /**
+   * Stops the tracking process.
+   */
+  stop(): void {
+    this.#poller.stop();
+  }
```

Comment:
In an ideal world, we will expose these methods onto the client, so that polling loops can be started/stopped in UI based way.

Meaning that we only start polling loops on components that need Multichain balances (like the asset list). However, if a user pops open the extension to sign a transaction, we don't need to start this polling loop, to minimize requests and improve perf.

Reply:
Let me get that working on a follow up PR